skglm/tests/test_group.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+import numpy as np
+from numpy.linalg import norm
+
+from skglm.penalties.block_separable import WeightedGroupL2
+from skglm.datafits.group import QuadraticGroup
+from skglm.solvers.group_bcd_solver import bcd_solver
+
+from skglm.utils import grp_converter, make_correlated_data
+from celer import GroupLasso, Lasso
+
+
+def _generate_random_grp(n_groups, n_features, shuffle=True):
+    all_features = np.arange(n_features, dtype=np.int32)
+    if shuffle:
+        np.random.shuffle(all_features)
+    splits = np.random.choice(all_features, size=n_groups+1,
+                              replace=False).astype(np.int32)
+    splits.sort()
+    splits[0], splits[-1] = 0, n_features
+
+    groups = [list(all_features[splits[i]: splits[i+1]])
+              for i in range(n_groups)]
+
+    return all_features, splits, groups   # grp_indices, grp_prt, groups
+
+
+@pytest.mark.parametrize("n_groups, n_features, shuffle",
+                         [[10, 50, True], [10, 50, False], [17, 53, False]])
```

Comment:
can one configuration be removed in order to have faster tests ? It seems they are redundant